src/datasets/vision/imagenet_reader/ImageNetReader.jl

```diff
@@ -0,0 +1,48 @@
+module ImageNetReader
+using ImageCore: channelview, colorview, AbstractRGB, RGB
+
+import ..FileDataset
+import ..read_mat
+import ..@lazy
+
+@lazy import JpegTurbo = "b835a17e-a41a-41e7-81f0-2f016b05efe0"
+
+const NCLASSES = 1000
+const IMGSIZE = (224, 224)
```

Comment:
these two could be part of metadata

Reply:
I've added `metadata["n_classes"]`. Should we make `imgsize` an argument to the `ImageNet` constructor or lock it to 224 x 224?

Reply:
That would be great. What happens if we don't pass `preferred_size` to JpegTurbo?

Reply:
According to https://github.com/JuliaML/MLDatasets.jl/pull/146#issuecomment-1164557500, this would slow down the image loading:
> If more control over the image loading is desired, like converting to a color upon reading or loading an image into a smaller size (much faster if it'll be downsized during training anyway) , one could also use JpegTurbo.jl directly